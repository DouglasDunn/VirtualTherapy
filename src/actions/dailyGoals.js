import database from '../firebase/firebase';

// ADD_DAILY_GOAL
export const addDailyGoal = (dailyGoal) => ({
  type: 'ADD_DAILY_GOAL',
  dailyGoal
});

export const startAddDailyGoal = (dailyGoalData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      goal = '',
      completed = false
    } = dailyGoalData;
    const dailyGoal = { goal, completed };

    return database.ref(`users/${uid}/dailyGoals`).push(dailyGoal).then((ref) => {
      dispatch(addDailyGoal({
        id: ref.key,
        ...dailyGoal
      }));
    });
  };
};

// SET_EXPENSES
export const setDailyGoals = (dailyGoals) => ({
  type: 'SET_DAILY_GOALS',
  dailyGoals
});

export const startSetDailyGoals = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/dailyGoals`).once('value').then((snapshot) => {
      const dailyGoals = [];

      snapshot.forEach((childSnapshot) => {
        dailyGoals.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });

      dispatch(setDailyGoals(dailyGoals));
    });
  };
};
