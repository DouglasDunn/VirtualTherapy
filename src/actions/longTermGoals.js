import database from '../firebase/firebase';

// ADD_LONG_TERM_GOAL
export const addLongTermGoal = (longTermGoal) => ({
  type: 'ADD_LONG_TERM_GOAL',
  longTermGoal
});

export const startAddLongTermGoal = (longTermGoalData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      goal = '',
      completed = false
    } = longTermGoalData;
    const longTermGoal = { goal, completed };

    return database.ref(`users/${uid}/longTermGoals`).push(longTermGoal).then((ref) => {
      dispatch(addLongTermGoal({
        id: ref.key,
        ...longTermGoal
      }));
    });
  };
};

// SET_LONG_TERM_GOALS
export const setLongTermGoals = (longTermGoals) => ({
  type: 'SET_LONG_TERM_GOALS',
  longTermGoals
});

export const startSetLongTermGoals = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/longTermGoals`).once('value').then((snapshot) => {
      const longTermGoals = [];

      snapshot.forEach((childSnapshot) => {
        longTermGoals.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });

      dispatch(setLongTermGoals(longTermGoals));
    });
  };
};
