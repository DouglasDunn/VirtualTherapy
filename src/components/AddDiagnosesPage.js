import React from 'react';
import { connect } from 'react-redux';
import DiagnosesForm from './DiagnosesForm';
// import { startAddExpense } from '../actions/expenses';

export class AddDiagnosesPage extends React.Component {
  // onSubmit = (expense) => {
  //   this.props.startAddExpense(expense);
  //   this.props.history.push('/');
  // };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <DiagnosesForm />
        </div>
      </div>
    );
  }
}

// <ExpenseForm
//   onSubmit={this.onSubmit}
// />

// const mapDispatchToProps = (dispatch) => ({
//   startAddExpense: (expense) => dispatch(startAddExpense(expense))
// });

export default AddDiagnosesPage;
//export default connect(undefined, mapDispatchToProps)(AddExpensePage);
