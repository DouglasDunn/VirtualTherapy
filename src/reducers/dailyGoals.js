// DailyGoals Reducer

const dailyGoalsReducerDefaultState = [];

export default (state = dailyGoalsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_DAILY_GOAL':
      return [
        ...state,
        action.dailyGoal
      ];
    case 'SET_DAILY_GOALS':
      return action.dailyGoals;
    default:
      return state;
  }
};
