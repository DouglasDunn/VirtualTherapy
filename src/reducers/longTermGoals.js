// LongTermGoals Reducer

const longTermGoalsReducerDefaultState = [];

export default (state = longTermGoalsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_LONG_TERM_GOAL':
      return [
        ...state,
        action.longTermGoal
      ];
    case 'SET_LONG_TERM_GOALS':
      return action.longTermGoals;
    default:
      return state;
  }
};
