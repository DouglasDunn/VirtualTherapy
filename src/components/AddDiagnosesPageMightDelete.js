import React from 'react';
import { connect } from 'react-redux';
import DiagnosesForm from './DiagnosesForm';
import { startAddDiagnoses } from '../actions/diagnoses';
import DiagnosesListItem from './DiagnosesListItem';

export class AddDiagnosesPage extends React.Component {
  onSubmit = (diagnoses) => {
    this.props.startAddDiagnoses(diagnoses);
    this.props.history.push('/diagnoses');
    console.log('add');
    console.log(this.props);
  };
  // nextPage = () => {
  //   this.props.history.push('/diagnoses');
  // };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <DiagnosesForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

// const mapStateToProps = (state) => ({
//   diagnoses: state.diagnoses
// });

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

// export default connect(mapStateToProps, mapDispatchToProps)(AddDiagnosesPage);

export default connect(undefined, mapDispatchToProps)(AddDiagnosesPage);

// <div className="content-container">
//   <div className="list-header">
//     <div className="show-for-mobile">Diagnoses</div>
//     <div className="show-for-desktop">Diagnoses</div>
//     <div className="show-for-desktop">Time</div>
//   </div>
//   <div className="list-body">
//     {
//       this.props.diagnoses.length === 0 ? (
//         <div className="list-item list-item--message">
//           <span>No Diagnoses yet, click the button above to add them.</span>
//         </div>
//       ) : (
//         this.props.diagnoses.map((diagnoses) => {
//           return <DiagnosesListItem key={diagnoses.id} {...diagnoses} />;
//         })
//       )
//     }
//   </div>
// </div>
// <div className="content-container">
//   <button onClick={this.nextPage} className="button">Next</button>
// </div>
