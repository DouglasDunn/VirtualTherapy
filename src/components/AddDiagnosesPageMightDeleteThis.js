import React from 'react';
import { connect } from 'react-redux';
import { startAddDiagnoses } from '../actions/diagnoses';

const initialErrorState = {
  diagnosedError: '',
  diagnosesListError: ''
};

export class AddDiagnosesPage extends React.Component {
  constructor(props) {
    super(props);
    let radioButton = 'no';
    let diagnosesArray;
    if (this.props.diagnoses) {
      if (Array.isArray(this.props.diagnoses)) {
        radioButton = 'yes';
        console.log(Array.isArray(this.props.diagnoses));
        console.log(this.props.diagnoses);
        diagnosesArray = this.props.diagnoses.map(({ diagnoses }) => diagnoses).join(', ');
      }
    }

    this.state = {
      diagnosed: this.props.diagnoses ? radioButton: '',
      diagnosesList: this.props.diagnoses ? diagnosesArray : '',
      ...initialErrorState
    };
  }

  onDiagnosedChange = (e) => {
    const diagnosed = e.target.value;
    this.setState(() => ({ diagnosed }));
  };

  onDiagnosesListChange = (e) => {
    const diagnosesList = e.target.value;
    this.setState(() => ({ diagnosesList }));
  };

  validate = () => {
    let diagnosedError = '';
    let diagnosesListError = '';

    if (!this.state.diagnosed) {
      diagnosedError = 'Please choose an option';
    }

    if ((this.state.diagnosed === 'yes') && !this.state.diagnosesList) {
      diagnosesListError = 'diagnoses list cannot be blank';
    } else {
      diagnosesListError = '';
    }

    if (diagnosedError || diagnosesListError) {
      this.setState({ diagnosedError, diagnosesListError });
      return false;
    }

    return true;
  };

  handlePrevious = () => {
    this.props.history.push('/add-drug-allergies');
  };

  onSubmit = (e) => {
    e.preventDefault();
    const isValid = this.validate();
    if (isValid) {
      if (this.state.diagnosed === 'no') {
        this.props.startAddDiagnoses({
          diagnosed: this.state.diagnosed,
          diagnosesList: ''
        });
      } else {
        console.log(this.state.diagnosesList);
        this.props.startAddDiagnoses({
          diagnosed: this.state.diagnosed,
          diagnosesList: this.state.diagnosesList
        });
      }
      this.props.history.push('/add-medication');
    }
  };

  renderDiagnosesList = () => {
    if (this.state.diagnosed === 'yes') {
      return (
        <div>
          {this.state.diagnosesListError && <p className="form__error">{this.state.diagnosesListError}</p>}
          <textarea
            placeholder="Type in all your symptoms and diagnoses separated by a comma"
            className="textarea"
            value={this.state.diagnosesList}
            onChange={this.onDiagnosesListChange}
          >
          </textarea>
        </div>
      );
    }
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <form className="form">
            {this.state.diagnosedError && <p className="form__error">{this.state.diagnosedError}</p>}
            Has a doctor ever diagnosed you with a mental health disease?
            <label>
              <input
                type="radio"
                checked={this.state.diagnosed === 'yes'}
                className="radio-input"
                value="yes"
                onChange={this.onDiagnosedChange}
              />
              Yes
            </label>
            <label>
              <input
                type="radio"
                checked={this.state.diagnosed === 'no'}
                className="radio-input"
                value="no"
                onChange={this.onDiagnosedChange}
              />
              No
            </label>
            {this.renderDiagnosesList()}
          </form>
          <div>
            <button onClick={this.handlePrevious} className="button">Previous</button>
            <button onClick={this.onSubmit} className="button">Next</button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  diagnoses: state.diagnoses
});

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddDiagnosesPage);
