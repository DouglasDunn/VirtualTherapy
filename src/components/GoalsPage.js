import React from 'react';
import { connect } from 'react-redux';
import GoalForm from './GoalForm';
import DailyGoalListItem from './DailyGoalListItem';
import LongTermGoalListItem from './LongTermGoalListItem';
import { startAddDailyGoal } from '../actions/dailyGoals';
import { startAddLongTermGoal } from '../actions/longTermGoals';

export class GoalsPage extends React.Component {
  onSubmitDailyGoal = (dailyGoal) => {
    this.props.startAddDailyGoal(dailyGoal);
  };
  onSubmitLongTermGoal = (longTermGoal) => {
    this.props.startAddLongTermGoal(longTermGoal);
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Goals</h1>
          </div>
        </div>
        <div className="content-container">
          <h2>Daily Goals</h2>
          <GoalForm
            onSubmit={this.onSubmitDailyGoal}
          />
          <div className="list-body">
            {
              this.props.dailyGoals.length === 0 ? (
                <div className="list-item list-item--message">
                  <span>No daily goals</span>
                </div>
              ) : (
                this.props.dailyGoals.map((dailyGoal) => {
                  return <DailyGoalListItem key={dailyGoal.id} {...dailyGoal} />;
                })
              )
            }
          </div>

          <h2>Long-Term Goals</h2>
          <GoalForm
            onSubmit={this.onSubmitLongTermGoal}
          />
          <div className="list-body">
            {
              this.props.longTermGoals.length === 0 ? (
                <div className="list-item list-item--message">
                  <span>No long-term goals</span>
                </div>
              ) : (
                this.props.longTermGoals.map((longTermGoal) => {
                  return <LongTermGoalListItem key={longTermGoal.id} {...longTermGoal} />;
                })
              )
            }
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    dailyGoals: state.dailyGoals,
    longTermGoals: state.longTermGoals
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddDailyGoal: (dailyGoal) => dispatch(startAddDailyGoal(dailyGoal)),
  startAddLongTermGoal: (longTermGoal) => dispatch(startAddLongTermGoal(longTermGoal))
});

export default connect(mapStateToProps, mapDispatchToProps)(GoalsPage);
