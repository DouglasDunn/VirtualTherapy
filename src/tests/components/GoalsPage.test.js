import React from 'react';
import { shallow } from 'enzyme';
import { GoalsPage } from '../../components/GoalsPage';
import dailyGoals from '../fixtures/dailyGoals';
import longTermGoals from '../fixtures/longTermGoals';

let startAddDailyGoal, startAddLongTermGoal;

beforeEach(() => {
  startAddDailyGoal = jest.fn();
  startAddLongTermGoal = jest.fn();
});

test('should render GoalsPage with dailyGoals and longTermGoals', () => {
  const wrapper = shallow(<GoalsPage dailyGoals={dailyGoals} longTermGoals={longTermGoals} startAddDailyGoal={startAddDailyGoal} startAddLongTermGoal={startAddLongTermGoal} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render GoalsPage with empty messages', () => {
  const wrapper = shallow(<GoalsPage dailyGoals={[]} longTermGoals={[]} startAddDailyGoal={startAddDailyGoal} startAddLongTermGoal={startAddLongTermGoal} />);
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmitDailyGoal', () => {
  const wrapper = shallow(<GoalsPage dailyGoals={dailyGoals} longTermGoals={longTermGoals} startAddDailyGoal={startAddDailyGoal} startAddLongTermGoal={startAddLongTermGoal} />);
  wrapper.find('GoalForm').at(0).prop('onSubmit')(dailyGoals[1]);
  expect(startAddDailyGoal).toHaveBeenLastCalledWith(dailyGoals[1]);
});

test('should handle onSubmitLongTermGoal', () => {
  const wrapper = shallow(<GoalsPage dailyGoals={dailyGoals} longTermGoals={longTermGoals} startAddDailyGoal={startAddDailyGoal} startAddLongTermGoal={startAddLongTermGoal} />);
  wrapper.find('GoalForm').at(1).prop('onSubmit')(longTermGoals[1]);
  expect(startAddLongTermGoal).toHaveBeenLastCalledWith(longTermGoals[1]);
});
