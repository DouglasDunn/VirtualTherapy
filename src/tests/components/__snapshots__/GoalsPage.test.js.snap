// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GoalsPage with dailyGoals and longTermGoals 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Goals
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <h2>
      Daily Goals
    </h2>
    <GoalForm
      onSubmit={[Function]}
    />
    <div
      className="list-body"
    >
      <DailyGoalListItem
        completed={false}
        goal="Run"
        id="1"
        key="1"
      />
      <DailyGoalListItem
        completed={true}
        goal="Drink water"
        id="2"
        key="2"
      />
      <DailyGoalListItem
        completed={false}
        goal="Eat vegetables"
        id="3"
        key="3"
      />
    </div>
    <h2>
      Long-Term Goals
    </h2>
    <GoalForm
      onSubmit={[Function]}
    />
    <div
      className="list-body"
    >
      <p>
        hi
      </p>
    </div>
  </div>
</div>
`;

exports[`should render GoalsPage with empty messages 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Goals
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <h2>
      Daily Goals
    </h2>
    <GoalForm
      onSubmit={[Function]}
    />
    <div
      className="list-body"
    >
      <div
        className="list-item list-item--message"
      >
        <span>
          No daily goals
        </span>
      </div>
    </div>
    <h2>
      Long-Term Goals
    </h2>
    <GoalForm
      onSubmit={[Function]}
    />
    <div
      className="list-body"
    >
      <div
        className="list-item list-item--message"
      >
        <span>
          No long-term goals
        </span>
      </div>
    </div>
  </div>
</div>
`;
